packages/workbox-webpack-plugin/src/lib/generate-manifest-with-webpack.js

```diff
@@ -27,11 +29,12 @@ const PRECACHE_MANIFEST_VAR = '__precacheManifest';
  *
  * @function generateManifestWithWebpack
  * @param {Array<module:workbox-build.ManifestEntry>} manifestEntries
- * @return {Promise<string>} service worker manifest file string
+ * @return {string} service worker manifest file string
  *
  * @private
  */
 module.exports = (manifestEntries) => {
-  const entriesJson = JSON.stringify(manifestEntries, null, 2);
-  return Promise.resolve(`self.${PRECACHE_MANIFEST_VAR} = ${entriesJson};`);
+  const sortedEntries = manifestEntries.sort((a, b) => a.url < b.url);
+  const entriesJson = stringify(sortedEntries, {space: 2});
```

Comment:
You might want to add a comment explaining that stringify is used here to remove any randomness from JSON.stringify making testing easier.

Reply:
I've added a comment. (The main motivation was to remove randomness so that the hashes were consistent, and the fact that testing also got easier was a nice side-effect.)